### ISSUE TITLE
can't use send function

### ISSUE THREAD

**Issue by mohammadshrahi** [blackhorse-one/stomp_dart_client#28]
Hello,
I'm trying to use WSS with my app and it's connected successfully.
but when i trying to send the server not receiving my message, also send function not giving any errors
` StompClient _client = StompClient(
        config: StompConfig(
            onConnect: onConnectCallback,
            connectionTimeout: Duration(minutes: 10),
            url: 'wss://URL/wathiq-ws',
            onDisconnect: (StompFrame frame) {
              print(frame);
            },
            onWebSocketDone: () {},
            onWebSocketError: (dynamic err) {
              print(err);
            },
            onStompError: (StompFrame frame) {
              print(frame.toString());
            }));`

`   client.send(
          destination: '/app/message/123',
          body: '{"text":"wss test","counter":"$counter"}');`

is there any suggestion to track this bug? and thanks in advance

**Comment by KammererTob** [comment]
Does your server support STOMP? You could listen to the `onDebugMessage` callback, which should be called with the serialized frame shortly before it is send via WebSocketChannel.

**Comment by mohammadshrahi** [comment]
yes, server support stomp, from javascript I'm trying to connect and send, both work correctly.
[Please find the HTML client here ](https://drive.google.com/file/d/197IfPISpFvr2_YZxOhHtL_0tIyqlqejg/view?usp=sharing)
onDebugMessage giving this message:
debug <<< CONNECTED
version:1.2
heart-beat:0,0
user-name:2ss34


debug <<< CONNECTED
version:1.2
heart-beat:0,0
user-name:2ss34


debug >>> SUBSCRIBE
id:sub-0
destination:/queue/message/123


debug >>> SEND
destination:
content-length:15

{"test":"test"}
by the way, subscribe callback not firing also. 
thanks in advance

**Comment by KammererTob** [comment]
Few things:
- in your HTML code you subscribe to /queue/message/* , not /queue/message/123
- it seems from the debug log that the destination of the send message is empty
Can you maybe show me a more complete code snippet of how you use this library? It should work as normal.

**Comment by mohammadshrahi** [comment]
/queue/message/* means send to all . also i try it and it didn't work for me . 
this is my full code .

`import 'dart:convert';
import 'dart:io';
import 'package:stomp_dart_client/stomp.dart';
import 'package:stomp_dart_client/stomp_config.dart';
import 'package:stomp_dart_client/stomp_frame.dart';

List<StompClient> _list = List();
double counter = 0;
int count = 1;
List<bool> finishList = List();
Future<void> main() async {
  // dart lib/providers/login/login_provider.dart
  start(_list);
}

Future<void> watcher(List<StompClient> _list) async {
  while (true) {
    print('in while');
    dynamic line = stdin.readLineSync(encoding: Encoding.getByName('utf-8'));
    print('enterChar$line');
    if (line == 'y') {
      for (int i = 0; i < count; i++) {
        _list[i].deactivate();
        exit(0);
      }
    } else if (line == 's') {
      send();
    } else if (line == 'sg') {
      sendGroup();
    } else if (line == 'st') {
      status();
    } else
      continue;
  }
}

void listn(StompClient stompClient) {
  stompClient.subscribe(
      destination: '/queue/message/*',
      callback: (frame) {
        print('reciving');
        print(frame.body);
      });
}

void status() {
  _list.forEach((client) {
    print(client.connected);
  });
}

Future<bool> start(List<StompClient> list) async {
  print('start');
  for (int i = 0; i < count; i++) {
    print('$i');
    StompClient _client = StompClient(
        config: StompConfig(
            onUnhandledReceipt: (frame) {
              print('onUnhandledReceipt');
            },
            onUnhandledMessage: (frame) {
              print('onUnhandledMessage');
            },
            onUnhandledFrame: (frame) {
              print('onUnhandledFrame');
            },
            onConnect: onConnectCallback,
            onDebugMessage: (string) {
              print('debug $string');
            },
            connectionTimeout: Duration(minutes: 10),
            url: 'wss://url/wathiq-ws?token=2ss34',
            onDisconnect: (StompFrame frame) {
              print(frame);
            },
            onWebSocketDone: () {},
            onWebSocketError: (dynamic err) {
              print(err);
            },
            onStompError: (StompFrame frame) {
              print(frame.toString());
            }));
    _client.activate();
    list.add(_client);
  }
  return true;
}

void onConnectCallback(StompClient client, StompFrame connectFrame) async {
  finishList.add(true);
  listn(client);
  print('connected ${client.connected}');

  if (finishList.length == count) {
    print('finish');
    watcher(_list);
  }
}

void send() {
  print('sending');
  print('meesage count$counter');
  try {
    _list.forEach((client) {
      counter++;
      print('meesage count$counter');
      client.send(
          headers: {}, destination: '/app/message/*', body: '{"test":"test"}');
    });
  } catch (e) {
    print(e);
  }
}

void sendGroup() {
  for (int i = 0; i < 1000; i++) {
    _list.forEach((client) {
      counter++;
      client.send(
          destination: '/app/message/*',
          body: '{"test":"wss test","counter":"$counter"}');
    });
  }
}
`

**Comment by KammererTob** [comment]
`_list` is always empty in the code above, so nothing will be send.

**Comment by mohammadshrahi** [comment]
`  start(_list); //im passing it to start function and i adding clients to it after `activate`
`
```
 _client.activate();
 list.add(_client);
```

**Comment by mohammadshrahi** [comment]
by the way, my code was working perfectly with the WS, but after updating our server to https, and I update to WSS, it stops working suddenly. 
thanks in advance

**Comment by mohammadshrahi** [comment]
updates:
i try to send message like this and it's work fine and i was able to receive in server side the message 
```
import 'dart:async';

import 'package:stomp_dart_client/sock_js/sock_js_parser.dart';
import 'package:stomp_dart_client/stomp_frame.dart';
import 'package:web_socket_channel/io.dart';

Future<void> main() async {
  IOWebSocketChannel channel;
  try {
    print('main');
    channel = IOWebSocketChannel.connect('wss://ramy.wiotschool.com/wathiq-ws');
    channel.stream.listen((dynamic event) {
      print('onData');
      print(event.toString());
    }, onDone: () {
      print('onDone');
    }, onError: (dynamic e) {
      print('onError');
    }, cancelOnError: false);
  } catch (e) {
    print('error' + e.toString());
  }
  SockJSParser _parser = SockJSParser(onStompFrame: (StompFrame frame) {
    print(frame.toString());
  }, onPingFrame: (StompFrame frame) {
    print(frame.toString());
  }, onDone: () {
    print('onDone');
  });

  // StompFrame frame = StompFrame(command: 'SEND', body: 'hello', headers: {});
  // print(_parser.serializeFrame(frame));

  channel.sink.add(
      "SEND\ndestination:/app/incident/123\ncontent-length:4\n\nramy\u0000");

  print('after listen');
}

```

**Comment by KammererTob** [comment]
Well from that information i would guess that your server was configured to use SockJS with Https instead of regular Websockets. I have just released a new version (0.3.6) which has a special constructor for SockJs. Please check the Readme and see if that helps.

**Comment by mohammadshrahi** [comment]
@KammererTob somehow the example code attached with the package it's work fine for me , and i couldn't differentiate
between my client and the example client so couldn't now what the issue exactly can u please close the issue for now and thanks for your supporting  👍
